### PR TITLE
[DB] Introduce EVO database

### PR DESCRIPTION
**Pull request by random-zebra**
Extracted from #2267.
This introduces the new database, used for all DMN-related things, and updates the dbWrapper.

Builds on top of:
- [x] #2269 
- [x] #2270

### CODE REVIEW

**Comment by random-zebra** [comment]
Rebased on master. Next in line for #2267

**Review by Fuzzbawls — APPROVED** [review]
Code ACK c27bbed57b638881d68778940e8917c1b965ee8b

File: src/dbwrapper.h [hunk 1]
```diff
@@ -207,12 +222,17 @@ class CDBWrapper
...
+    template <typename K>
+    bool ReadDataStream(const K& key, CDataStream& ssValue) const
     {
         CDataStream ssKey(SER_DISK, CLIENT_VERSION);
         ssKey.reserve(DBWRAPPER_PREALLOC_KEY_SIZE);
         ssKey << key;
+        return ReadDataStream(ssKey, ssValue);
+    }
+
+    bool ReadDataStream(const CDataStream& ssKey, CDataStream& ssValue) const
```
Comment (furszy) [inline src/dbwrapper.h:243]: I'm not sure how this is working.

So, i'm seeing the following flow: `Read(const K& key, V& value)` calls `Read(const CDataStream& ssKey, V& value)` which calls `ReadDataStream(const CDataStream& ssKey, CDataStream& ssValue`.

Inside the last function the `ssValue` of `ReadDataStream` is never assigned nor used inside the function. Line 227 is creating a new object instead. So when it get back to the first `Read`, `ssValue` has no value, so `value` is not assigned.

Comment (furszy) [reply src/dbwrapper.h:243]: ok, found why it's working. You solved it in a later commit f259774db579dd03b22307d3b069a22c10975c70 .

**Review by furszy — APPROVED** [review]
Code ACK c27bbed.

Have to add that spent some time thinking about this and will return to this area after finishing with #2267, so we can talk and brainstorm more about it after reviewing the complete set of changes.

**Pull request merged**